modules/platforms/cpp/network/src/network/ssl/ssl_gateway.cpp

```diff
@@ -66,77 +66,114 @@ namespace ignite
-                    ssleay32 = LoadSslLibrary("ssleay32");
-                }
+                bool isLoaded = false;
 
-                if (!libssl.IsLoaded() && (!libeay32.IsLoaded() || !ssleay32.IsLoaded()))
-                {
-                    if (!libssl.IsLoaded())
-                        throw IgniteError(IgniteError::IGNITE_ERR_GENERIC,
-                            "Can not load neccessary OpenSSL library: libssl");
+                if (!home.empty())
+                    isLoaded = TryLoadSslLibraries(home);
 
+                // Try load from system path.
+                if (!isLoaded)
+                    isLoaded = TryLoadSslLibraries("");
+
+                if (!isLoaded)
+                {
+#ifdef _WIN32
                     std::stringstream ss;
 
-                    ss << "Can not load neccessary OpenSSL libraries:";
+                    ss << "Can not load necessary OpenSSL libraries:";
+
+                    if (!libssl.IsLoaded() || !libcrypto.IsLoaded())
+                    {
+                        if (!libssl.IsLoaded())
+                            ss << " libssl";
 
-                    if (!libeay32.IsLoaded())
-                        ss << " libeay32";
+                        if (!libcrypto.IsLoaded())
+                            ss << " libcrypto";
+                    }
+                    else
+                    {
+                        if (!libeay32.IsLoaded())
+                            ss << " libeay32";
 
-                    if (!ssleay32.IsLoaded())
-                        ss << " ssleay32";
+                        if (!ssleay32.IsLoaded())
+                            ss << " ssleay32";
+                    }
 
                     std::string res = ss.str();
 
                     throw IgniteError(IgniteError::IGNITE_ERR_GENERIC, res.c_str());
+#else
+                    if (!libssl.IsLoaded())
+                        throw IgniteError(IgniteError::IGNITE_ERR_GENERIC,
+                            "Can not load necessary OpenSSL library: libssl");
+#endif
                 }
             }
 
+            bool SslGateway::TryLoadSslLibraries(const std::string& homeDir) {
+#ifdef _WIN32
+#ifdef _WIN64
+                libcrypto = LoadSslLibrary("libcrypto-3-x64", homeDir);
+                libssl = LoadSslLibrary("libssl-3-x64", homeDir);
+
+                if (!libssl.IsLoaded() || !libcrypto.IsLoaded()) {
+                    libcrypto = LoadSslLibrary("libcrypto-1_1-x64", homeDir);
+                    libssl = LoadSslLibrary("libssl-1_1-x64", homeDir);
+                }
+#else
+                libcrypto = LoadSslLibrary("libcrypto-3", homeDir);
+                libssl = LoadSslLibrary("libssl-3", homeDir);
+
+                if (!libssl.IsLoaded() || !libcrypto.IsLoaded()) {
+                    libcrypto = LoadSslLibrary("libcrypto-1_1", homeDir);
+                    libssl = LoadSslLibrary("libssl-1_1", homeDir);
+                }
+#endif
```

Comment:
This can probably reduce code duplication. Anyway, just a suggestion, it is up for you to decide.
```suggestion
#ifdef _WIN64
#define SSL_LIB_POSTFIX "-x64"
#else
#define SSL_LIB_POSTFIX ""
#endif
                libcrypto = LoadSslLibrary("libcrypto-3" SSL_LIB_POSTFIX, homeDir);
                libssl = LoadSslLibrary("libssl-3" SSL_LIB_POSTFIX, homeDir);

                if (!libssl.IsLoaded() || !libcrypto.IsLoaded()) {
                    libcrypto = LoadSslLibrary("libcrypto-1_1" SSL_LIB_POSTFIX, homeDir);
                    libssl = LoadSslLibrary("libssl-1_1" SSL_LIB_POSTFIX, homeDir);
                }
```

Reply:
+1 Good idea, implemented